packages/core-ethereum/src/types/ticket.ts

```diff
@@ -89,39 +83,52 @@ class Ticket extends Uint8ArrayE implements Types.Ticket {
   }
 
   get epoch(): UINT256 {
-    return new UINT256(new Uint8Array(this.buffer, this.epochOffset, EPOCH_SIZE))
+    return new UINT256(new BN(new Uint8Array(this.buffer, this.epochOffset, UINT256.SIZE)))
   }
 
   get amountOffset(): number {
-    return this.byteOffset + Address.SIZE + Hash.SIZE + EPOCH_SIZE
+    return this.byteOffset + Address.SIZE + Hash.SIZE + UINT256.SIZE
   }
 
   get amount(): Balance {
-    return Balance.fromUint96(new Uint8Array(this.buffer, this.amountOffset, 12))
+    return new Balance(new BN(new Uint8Array(this.buffer, this.amountOffset, Balance.SIZE)))
   }
 
   get winProbOffset(): number {
-    return this.byteOffset + Address.SIZE + Hash.SIZE + EPOCH_SIZE + UINT96_SIZE
+    return this.byteOffset + Address.SIZE + Hash.SIZE + UINT256.SIZE + UINT256.SIZE
   }
 
   get winProb(): Hash {
     return new Hash(new Uint8Array(this.buffer, this.winProbOffset, Hash.SIZE))
   }
 
   get channelIterationOffset(): number {
-    return this.byteOffset + Address.SIZE + Hash.SIZE + EPOCH_SIZE + UINT96_SIZE + Hash.SIZE
+    return this.byteOffset + Address.SIZE + Hash.SIZE + UINT256.SIZE + UINT256.SIZE + Hash.SIZE
   }
 
   get channelIteration(): UINT256 {
-    return new UINT256(new Uint8Array(this.buffer, this.channelIterationOffset, EPOCH_SIZE))
+    return new UINT256(new BN(new Uint8Array(this.buffer, this.channelIterationOffset, UINT256.SIZE)))
   }
 
   get hash(): Promise<Hash> {
-    return Promise.resolve(toEthSignedMessageHash(u8aToHex(this)))
+    return Promise.resolve(
+      toEthSignedMessageHash(
```

Comment:
the order of the items needs to be the same as the one used in the SC

Reply:
should we add a comment to this effect?

Reply:
added https://github.com/hoprnet/hoprnet/pull/833/commits/d1f07b8a5ad6222ceeba44fc7925e1bf11253c71